DEVELOPMENTGUIDE.md

```diff
@@ -0,0 +1,78 @@
+# Development Guide
+The Development Guide establishes conventions that regular contributors are expected to follow on ScubaGoggles. If you are a new or first-time contributor, you are not required to follow all these conventions, though it is certainly appreciated. See instead [CONTRIBUTING.md](CONTRIBUTING.md) for licencing details and guidance for new contributors.
+
+## Branching Strategy
+We adhere to [GitHub flow](https://docs.github.com/en/get-started/using-github/github-flow) as closely as our specific constraints allow. In general, this means:
+- Development is performed on short-lived features branches (see [Contributing Code](#contributing-code) for more details).
+- There is only one primary branch, `main`. After review, changes are committed directly to main, rather than to long-lived develop or release branches.
+
+There are exceptions to this, necessitated by the extensive review process the baseline documents themselves undergo, during which no baseline changes can be made.
+To accommodate this, the development lifecycle on ScubaGoggles is as follows:
+![image](/docs/images/scubagoggles_lifecycle.svg)
+
+
+Baseline submission triggers a baseline freeze, at which point the following actions need to be performed:
+1. Bump the code in `main` to the upcoming version (to reduce merge conflicts downstream).
+2. Make a note of the last commit hash, to help in case of complications rebasing down the line.
+3. Create a temporary protected branch dedicated to baseline changes.
+
+A new release triggers the transition back to normal operations. At this time, the following actions need to be performed:
+1. Pause development on main and perform functional testing.
+2. Assuming the testing results are satisfactory, create the release off of `main`.
+3. Rebase the temporary baseline branch with `main`.
+4. Implement the baseline code issues created during the baseline freeze to get the release branch into a fully functional state.
+5. Merge the baseline branch into `main`.
+6. Delete the baseline branch.
+
+## Branching Structure
+See [Branching Structure Diagram](BRANCHINGSTRUCTURE.md) for a visual representation of the branch structure, both during a code freeze and otherwise.
+
+## Creating Issues
+All needed changes (e.g., new features, bug fixes) need to be tracked in an issue. When creating an issue:
+- Be sure to follow the template.
+- Add the appropriate labels (e.g., "bug," "enhancement").
```

Comment:
For new issues: `Leave the assignee option blank or assign yourself if you plan to implement the issue` ?

Reply:
I think my vote would be that you not assign yourself until you're actively starting to work the issue. Just want to avoid the situation where someone wants to handle and issue, assigns themselves, but then something comes up and they never get to it, leaving the issue in limbo.

Reply:
> 4. Leave the milestone option blank...

Done

Reply:
> I think my vote would be that you not assign yourself until you're actively starting to work the issue. Just want to avoid the situation where someone wants to handle and issue, assigns themselves, but then something comes up and they never get to it, leaving the issue in limbo.

Sounds good